Open Judge System/Services/OJS.Services/Business/ParticipantScores/ParticipantScoresBusinessService.cs

```diff
@@ -262,8 +265,34 @@ private string GetSubmissionKey(StatisticsLevel level, Submission submission)
 
         private int GetSubmissionCodeLength(Submission submission)
             => submission.IsBinaryFile
-                ? 0
-                : Regex.Matches(submission.ContentAsString, "\n").Count + 1;
+                ? this.GetFileSubmissionLength(submission.Content)
+                : this.GetCodeSubmissionLength(submission.ContentAsString);
+
+        private int GetFileSubmissionLength(byte[] submissionFileContent)
+        {
+            var zip = new ZipArchive(new MemoryStream(submissionFileContent), ZipArchiveMode.Read, true);
+            var eligibleFileEntries = zip
+                .Entries
+                .Where(e =>
+                    GlobalConstants.ParticipationStatisticsFileSubmissionsAllowedExtensions
+                        .Any(fe => e.FullName.EndsWith(fe)));
+
+            var sumFileContentsLength = eligibleFileEntries
+                .Select(e =>
+                {
+                    var fileContent = new StreamReader(e.Open(), Encoding.UTF8).ReadToEnd();
+                    
+                    return this.GetCodeSubmissionLength(fileContent);
+                })
+                .Sum();
+            
+            zip.Dispose();
```

Comment:
Instead of disposing manually you can use - using var zip = new ZipArchive(new MemoryStream(submissionFileContent), ZipArchiveMode.Read, true);
Also, you can use the stream into using too.